src/payload.rs

```diff
@@ -174,6 +228,16 @@ pub fn to_json_error_body(message: &str, request_id: Option<Vec<u8>>) -> String
     .expect("JSON error object")
 }
 
+pub fn urlencoded_bytes_to_map(input: &[u8]) -> serde_json::Map<String, serde_json::Value> {
+    let mut map = serde_json::Map::new();
```

Comment:
nit: we have the input size, so we can use the `Map:with_capacity()` constructor:

```suggestion
    let mut map = serde_json::Map::with_capacity(input.len());
```